### DISCUSSION TITLE
Access restrictions specified in the lexicon

### DISCUSSION THREAD

**Discussion by byoda** [bluesky-social/atproto#2704] [Protocol (atproto)]
Hi,

I'm reviewing atproto and the BSky PDS to the [BYODA pds](https://github.com/byoda/byoda-python) I wrote and that is used for the [BYO.Tube](https://byo.tube) site. There is an uncanny number of similarities between the two in both architecture and implementation! One thing I haven't been able to find yet is whether you can specify access restrictions in a lexicon, for use cases like: 
1. this post, comment or video should only be visible and accessible to people I have linked to in my social graph as 'friends'.
2. I want to grant a 3rd-party app access to this subset of data stored in my pds.

Did I miss this in the atproto spec or is it not supported?

**Comment by snarfed** [comment]
Short answer is, this isn't supported. Right now ATProto is a fully public protocol and network: https://bsky.social/about/blog/5-19-2023-user-faq . There are off-protocol exceptions like chat, though, and they're planning to add E2EE messaging in the future.
